From 3002099b6d5ce6e5a74f4d0a5bea49c865949234 Mon Sep 17 00:00:00 2001
From: Brian Bockelman <bbockelman@morgridge.org>
Date: Mon, 28 Feb 2022 22:51:59 -0600
Subject: [PATCH 15/17] Use gridmap.name as a flag - no need to preserve the
 actual name (as this is also in the Entity name).

(cherry picked from commit 4ede2baf8e9e6af9131caa4cfd3987bd0dcad264)
---
 src/XrdHttp/XrdHttpSecurity.cc     | 2 +-
 src/XrdSecgsi/XrdSecProtocolgsi.cc | 2 +-
 src/XrdVoms/XrdVomsMapfile.cc      | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/xrootd/src/XrdHttp/XrdHttpSecurity.cc b/xrootd/src/XrdHttp/XrdHttpSecurity.cc
index e187851c0..638045eb6 100644
--- a/xrootd/src/XrdHttp/XrdHttpSecurity.cc
+++ b/xrootd/src/XrdHttp/XrdHttpSecurity.cc
@@ -165,7 +165,7 @@ XrdHttpProtocol::HandleGridMap(XrdLink* lp)
       TRACEI(DEBUG, " Mapping name: '" << SecEntity.moninfo << "' --> " << bufname);
       if (SecEntity.name) free(SecEntity.name);
       SecEntity.name = strdup(bufname);
-      SecEntity.eaAPI->Add("gridmap.name", bufname, true);
+      SecEntity.eaAPI->Add("gridmap.name", "1", true);
     }
     else {
       TRACEI(ALL, " Mapping name: " << SecEntity.moninfo << " Failed. err: " << mape);
diff --git a/xrootd/src/XrdSecgsi/XrdSecProtocolgsi.cc b/xrootd/src/XrdSecgsi/XrdSecProtocolgsi.cc
index e93deb912..c7bcb1d7e 100644
--- a/xrootd/src/XrdSecgsi/XrdSecProtocolgsi.cc
+++ b/xrootd/src/XrdSecgsi/XrdSecProtocolgsi.cc
@@ -1954,7 +1954,7 @@ int XrdSecProtocolgsi::Authenticate(XrdSecCredentials *cred,
                DEBUG("user mapping lookup successful: name is '"<<name<<"'");
             }
             Entity.name = strdup(name.c_str());
-            Entity.eaAPI->Add("gridmap.name", name.c_str(), true);
+            Entity.eaAPI->Add("gridmap.name", "1", true);
          }
       }
       // If not set, use DN
diff --git a/xrootd/src/XrdVoms/XrdVomsMapfile.cc b/xrootd/src/XrdVoms/XrdVomsMapfile.cc
index 254ac3d98..824edd94a 100644
--- a/xrootd/src/XrdVoms/XrdVomsMapfile.cc
+++ b/xrootd/src/XrdVoms/XrdVomsMapfile.cc
@@ -263,7 +263,7 @@ XrdVomsMapfile::Apply(XrdSecEntity &entity)
     // set when the mapfile is used to generate the name.
     std::string gridmap_name;
     auto gridmap_success = entity.eaAPI->Get("gridmap.name", gridmap_name);
-    if (gridmap_success) {
+    if (gridmap_success && gridmap_name == "1") {
         return 0;
     }
 
-- 
2.25.1

